src/models/graph.ts

```diff
@@ -0,0 +1,457 @@
+
+import { Record as ImmuRecord, Map as ImmuMap } from "immutable";
+import { OSCQueryRNBOInstance, OSCQueryRNBOInstanceConnections, OSCQueryRNBOInstancePresetEntries, OSCQueryRNBOJackPortInfo } from "../lib/types";
+import { ParameterRecord } from "./parameter";
+import { MessageInportRecord, MessageOutputRecord } from "./messages";
+import { PresetRecord } from "./preset";
+
+export enum ConnectionType {
+	Audio = "audio",
+	MIDI = "midi"
+}
+
+export enum PortDirection {
+	Sink,
+	Source
+}
+
+export enum NodeType {
+	Patcher = "patcher",
+	System = "system"
```

Comment:
probably eventually wanna add "other"

Reply:
Guess we can cross the bridge once we get there.